commands.go

```diff
@@ -152,6 +152,13 @@ type Cmdable interface {
 	SRem(key string, members ...interface{}) *IntCmd
 	SUnion(keys ...string) *StringSliceCmd
 	SUnionStore(destination string, keys ...string) *IntCmd
+	XAdd(key string, id string, els ...interface{}) *StringCmd
+	XAddMaxLen(key string, id string, maxLen int64, approximate bool, els ...interface{}) *StringCmd
```

Comment:
I am almost sure that `XAdd` will be extended later, .e.g. with MaxAge option https://github.com/antirez/redis/issues/4450. Also `redis.XAddMaxLen("stream", "12345.0", 100, false, "foo", "bar")` is not very readable. Can we change this to:

```
redis.XAddExt("stream", "12345.1", &redis.XAdd{
    MaxLen: 100,
    Approximate: true,
    MaxAge: time.Hour, // can be added later
}, "foo", "bar")
```